File: skills/public/code-business-dag-analysis-pipeline/evals/fixtures/video_pipeline_missing_decode.py
import cv2


class Detector:
    def predict(self, frame):
        return [{"box": [10, 20, 80, 120], "score": 0.91}]


class Tracker:
    def update(self, detections):
        return [{"track_id": 1, "box": item["box"]} for item in detections]


def save_tracks(tracks, output_path):
    with open(output_path, "w", encoding="utf-8") as file:
        for track in tracks:
            file.write(f"{track['track_id']},{track['box']}\n")


def run_pipeline(video_path, output_path):
    capture = cv2.VideoCapture(video_path)
    detector = Detector()
    tracker = Tracker()

    ok, frame = capture.read()
    if not ok:
        return []

    detections = detector.predict(frame)
    tracks = tracker.update(detections)
    save_tracks(tracks, output_path)
    return tracks

File: skills/public/code-business-dag-analysis-pipeline-cpp/evals/fixtures/video_pipeline_missing_decode.cpp
#include <opencv2/opencv.hpp>
#include <fstream>

int main() {
    cv::VideoCapture capture(0);
    cv::Mat frame;
    capture.read(frame);

    auto detections = detector.predict(frame);
    auto tracks = tracker.update(detections);

    std::ofstream out("tracks.txt");
    out.write((char*)&tracks, sizeof(tracks));
    return 0;
}
